src/excited_workflow/carbon_tracker.py

```diff
@@ -0,0 +1,266 @@
+    df_reduced = df_pycaret[::10]
+
+    return df_reduced
+
+
+def train_model(
+    df: pd.DataFrame, number: int, x_keys: list[str], y_key: str
+) -> tuple[xr.Dataset, xr.Dataset]:
+    """Train a model on training data and create prediction.
+
+    Args:
+        df: dataframe which needs to be converted for training.
+        number: number of groups.
+        x_keys: list of input variables.
+        y_key: target variable name.
+
+    Returns:
+        Dataset for training and prediction dataset.
+    """
+    mask = df["group"] != number
+    df_train = df[mask]
+    df_reduced = create_df(df_train, x_keys, y_key)
+
+    pycs = pycaret.regression.setup(df_reduced, target=y_key, verbose=False)
+    model = pycs.compare_models(
+        include=["lightgbm"], n_select=1, round=1, cross_validation=False
+    )
+
+    df_prediction = df[~mask]
+    data = create_df(df_prediction, x_keys, y_key)
+    ds_target = xr.Dataset.from_dataframe(data)
+    data.drop(y_key, axis=1, inplace=True)
+
+    prediction = pycs.predict_model(model, data=data)
+    ds_prediction = xr.Dataset.from_dataframe(prediction)
+
+    return ds_target, ds_prediction
+
+
+def calculate_rmse(prediction: xr.DataArray, target: xr.DataArray) -> np.ndarray:
+    """Calculate RMSE.
+
+    Args:
+        prediction: column of prediction.
+        target: column of target.
+
+    Returns:
+        RMSE
+    """
+    rmse = np.sqrt(((prediction - target) ** 2).mean(dim="time", skipna=True))
+    return rmse
+
+
+def create_scatterplot(prediction: xr.DataArray, target: xr.DataArray) -> None:
+    """Create scatterplot of prediction vs. target.
+
+    Args:
+        prediction: dataframe column of prediction.
+        target: dataframe column of target variable.
+
+    Returns:
+        Scatterplot of prediction vs. target.
+    """
+    plt.scatter(prediction, target)
+    plt.xlabel("Prediction")
+    plt.ylabel("Target")
+
+
+def validate_model(
+    ds: xr.Dataset, groups: int, x_keys: list[str], y_key: str, output_path: Path
+) -> None:
+    """Validate the trained model by calculating rmse and scatterplots.
+
+    Args:
+        ds: dataset for training.
+        groups: number of groups.
+        x_keys: list of input variables.
+        y_key: target variable name.
+        output_path: directory to output rmse and scatterplots.
+    """
+    time = datetime.datetime.now().strftime("%Y-%m-%d_%H")
+    output_dir = output_path / f"carbon_tracker-{time}"
+    output_dir.mkdir(parents=True, exist_ok=True)
+
+    df_group = create_groups(ds, groups)
+
+    for group in range(groups):
+        target_ds, prediction = train_model(df_group, group, x_keys, y_key)
+        rmse = calculate_rmse(prediction["prediction_label"], target_ds[y_key])
+        rmse.to_netcdf(output_dir / f"rmse{group}.nc")
+        create_scatterplot(prediction["prediction_label"], target_ds[y_key])
+        plt.savefig(output_dir / f"scatter{group}.png")
+        plt.close()
+
+
+def save_model(
```

Comment:
Splitting this function up in training and saving is better I think

Reply:
It could be nice to make a generic train function, with an optional argument "group". This way the training function can be reused for the CV and the final model training.

Reply:
Added a training function but I am unsure it works in the way you meant.